amplify-tools-gradle-plugin/src/main/groovy/com/amplifyframework/tools/gradle/plugin/AmplifyTools.groovy

```diff
@@ -0,0 +1,219 @@
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+class AmplifyTools implements Plugin<Project> {
+    void apply(Project project) {
+        def doesNodeExist = true
+        def doesGradleConfigExist
+
+        def profile = 'default'
+        def accessKeyId = null
+        def secretAccessKey = null
+        def region = null
+        def envName = 'amplify'
+        def syncEnabled = 'true'
+
+        project.task('verifyNode') {
+            try {
+                project.exec {
+                    commandLine 'which', 'node'
+                    standardOutput = new ByteArrayOutputStream()
+                }
+            } catch (e) {
+                doesNodeExist = false
+                println("Node is not installed. Visit https://nodejs.org/en/download/ to install it")
+            }
+        }
+
+        project.task('createAmplifyApp') {
+            doesGradleConfigExist = project.file('amplify-gradle-config.json').isFile()
```

Comment:
How is this file created and managed? Also, by storing it in the project directory, we increase the risk of a customer accidentally committing their AWS account credentials to source control.

Reply:
This file is created through amplify-app - https://github.com/aws-amplify/amplify-cli/blob/b38deb59223417c67d41838c7af21fe4184babd0/packages/amplify-app/src/index.js#L351

The file is added to the gitignore so it would have to be manually removed from there to be commited

Reply:
Adding to .gitignore is a brittle solution; it's possible to override that with a `git add --force <path that accidentally includes this file even though you didn't actually intend to add it>` and then you're back to dealing with committed credentials. It sounds like this is something we need to discuss with the CLI team.

Reply:
Related to this, when testing the tools in iOS yesterday I realized that we rely on the top-level `.gitignore` to omit some of the files inside the `amplify/` folder. It seems to me like that `amplify/` folder should have self-contained `.gitignore` rules that don't depend on the top-level file having the right config so we don't delegate that responsibility to customers.

Reply:
I think this warrants further discussion but the change can't be made here. Could we open up an issue regarding ```.gitignore``` on the CLI repo so that we can continue the conversation there?